lwpreflight/azure/policy.go

```diff
@@ -0,0 +1,70 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization"
+)
+
+func FetchPolicies(p *Preflight) error {
+	if p.caller.IsAdmin {
+		return nil
+	}
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create credential: %v", err)
+	}
+
+	// Get role assignments for the caller
+	client, err := armauthorization.NewRoleAssignmentsClient(p.subscriptionID, cred, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create role assignments client: %v", err)
```

Comment:
nit: would the original `err` contain enough information about the error? It might be better to just do `return err`. Same for the errors below.